### PR TITLE
[14.0][IMP] account_payment_order: Add hook method for finding previous payment orders

### PR DESCRIPTION
**Pull request by AmmarOfficewalaSerpentCS**
Add hook method for finding previous payment orders

### CODE REVIEW

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please document the purpose of the improvement.

The  method you are adding should be private, and the usual practice is to return the domain to search, not directly the search, as it's more flexible.

**Comment by AmmarOfficewalaSerpentCS** [comment]
Hello @pedrobaeza , 
I develop a new module for a payment order 
In this module for only this search, I need to overwrite all methods in my new module  that's why I improved this code
Here it is a PR link :- https://github.com/OCA/bank-payment/pull/831

**Comment by AmmarOfficewalaSerpentCS** [comment]
https://github.com/OCA/bank-payment/pull/831

**Comment by pedrobaeza** [comment]
You don't have to tell it to me, but document it:

- Proper commit message.
- Proper PR title.
- Proper main comment.

**Comment by pedrobaeza** [comment]
The commit message should be:

```
[IMP] account_payment_order: Add hook method for finding previous payment orders
```

File: account_payment_order/models/account_move.py [hunk 1]
```diff
@@ -81,11 +84,7 @@ def create_account_payment_line(self):
                 raise UserError(_("No Payment Mode on invoice %s") % move.name)
             for payment_mode in payment_modes:
                 payorder = apoo.search(
-                    [
-                        ("payment_mode_id", "=", payment_mode.id),
-                        ("state", "=", "draft"),
-                    ],
-                    limit=1,
+                    self.get_account_payment_domain(payment_mode.id), limit=1
```
Comment (pedrobaeza) [inline account_payment_order/models/account_move.py]: Don't pass ID, but record

**Comment by pedrobaeza** [comment]
`[14.0]` part should be out on the commit message. That's only for the PR message, but when seeing commit history, the version is not adding useful information.

**Review by dreispt — APPROVED** [review]
The change is simple, and follows the common practice to prepare a domain in a separate (and extendable) method.

**Comment by pedrobaeza** [comment]
Travis is failing though. Not sure why.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Let's try the merge:

/ocabot merge patch

**Pull request merged**